Frameworks/CoreImage/CIImage.mm

```diff
@@ -73,90 +74,164 @@ @implementation CIImage
 /**
  @Status Interoperable
 */
-- (CIImage*)init {
++ (CIImage*)emptyImage {
+    return [[[CIImage alloc] init] autorelease];
+}
+
+/**
+ @Status Interoperable
+*/
+- (CGRect)extent {
+    return self->_extent;
+}
+
+/**
+ @Status Interoperable
+*/
+- (instancetype)init {
     if (self = [super init]) {
-        _cgImage = nil;
-        _color = nil;
-        _filter = nil;
         _extent = CGRectInfinite;
     }
     return self;
 }
 
+- (instancetype)_initWithCGImage:(CGImageRef)image color:(CIColor*)color {
+    if (self = [self initWithCGImage:image]) {
+        _color.attach([static_cast<id>(color) copy]);
+    }
+
+    return self;
+}
+
+/**
+ @Status Interoperable
+*/
+- (instancetype)initWithColor:(CIColor*)color {
+    if (!color) {
+        [self release];
+        return nil;
+    }
+
+    if (self = [self init]) {
+        _color.attach([static_cast<id>(color) copy]);
+    }
+
+    return self;
+}
+
+/**
+ @Status Interoperable
+*/
+- (instancetype)initWithCGImage:(CGImageRef)cgImage {
+    if (!cgImage) {
+        [self release];
+        return nil;
+    }
+
+    if (self = [self init]) {
+        _cgImage.reset(CGImageCreateCopy(cgImage));
```

Comment:
Sorry, *not* mutable.

---
In reply to: [94636453](https://github.com/Microsoft/WinObjC/pull/1625#discussion_r94636453) [](ancestors = 94636453)